Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedMapModel.swift

```diff
@@ -16,18 +16,86 @@ import SwiftUI
 import ArcGIS
 
 /// An object that encapsulates state about a preplanned map.
-public class PreplannedMapModel: ObservableObject, Identifiable {
+@MainActor
+class PreplannedMapModel: ObservableObject, Identifiable {
     /// The preplanned map area.
     let preplannedMapArea: PreplannedMapArea
     
-    /// The result of the download job. When the result is `.success` the mobile map package is returned.
-    /// If the result is `.failure` then the error is returned. The result will be `nil` when the preplanned
-    /// map area is still packaging or loading.
-    @Published private(set) var result: Result<MobileMapPackage, Error>?
+    /// The combined status of the preplanned map area.
+    @Published private(set) var status: Status = .notLoaded
     
-    init(
-        preplannedMapArea: PreplannedMapArea
-    ) {
+    init(preplannedMapArea: PreplannedMapArea) {
         self.preplannedMapArea = preplannedMapArea
+        
+        // Kick off a load of the map area.
+        Task.detached { await self.load() }
+    }
+    
+    /// Loads the preplanned map area and updates the status.
+    private func load() async {
+        do {
+            // Load preplanned map area to obtain packaging status.
+            status = .loading
+            try await preplannedMapArea.retryLoad()
+            // Note: Packaging status is `nil` for compatibility with
+            // legacy webmaps that have incomplete metadata.
+            // If the area loads, then you know for certain the status is complete.
+            updateStatus(for: preplannedMapArea.packagingStatus ?? .complete)
+        } catch is IllegalStateError {
```

Comment:
```suggestion
        } catch is MappingError {
```
When `retryLoad()` throws an error due to the map area packaging, the error is caught in the last catch block as a MappingError. This error should be cast as a MappingError instead.

Reply:
which mapping error is it? we should be specific about it

Reply:
This is the mapping error:
```
(lldb) po error
▿ MappingError
  ▿ packagingNotComplete : 1 element
    - details : "Failed to load preplanned map as the packaging status is not complete. Status: processing."
(lldb) po error.localizedDescription
"The operation couldn’t be completed. (ArcGIS.MappingError error 20.)"
```